guillotina/contrib/catalog/pg.py

```diff
@@ -0,0 +1,549 @@
+            operator = '?|'
+            field = field[:-len('__in')]
+        elif field.endswith('__eq'):
+            operator = '='
+            field = field[:-len('__eq')]
+        elif field.endswith('__gt'):
+            operator = '>'
+            field = field[:-len('__gt')]
+        elif field.endswith('__lt'):
+            operator = '<'
+            field = field[:-len('__lt')]
+        elif field.endswith('__gte'):
+            operator = '>='
+            field = field[:-len('__gte')]
+        elif field.endswith('__lte'):
+            operator = '<='
+            field = field[:-len('__lte')]
+
+        index = get_index_definition(field)
+        if index is None:
+            return None
+
+        _type = index['type']
+        if _type in _type_mapping:
+            try:
+                result = _type_mapping[_type](value)
+            except ValueError:
+                # invalid, can't continue... We could throw query parse error?
+                return None
+        elif _type == 'date':
+            result = parse(value).replace(tzinfo=None)
+        elif _type == 'boolean':
+            if value in ('true', 'True', 'yes', '1'):
+                result = True
+            else:
+                result = False
+        elif _type == 'keyword' and operator not in ('?', '?|'):
+            operator = '?'
+
+        if operator == '?|':
+            result = to_list(value)
+
+        if operator == '?' and isinstance(result, list):
+            operator = '?|'
+
+        pg_index = get_pg_index(field)
+        return pg_index.where(result, operator), result, pg_index.select()
+
+    def __call__(self, params: typing.Dict) -> ParsedQueryInfo:
+        query_info = super().__call__(params)
+
+        wheres = []
+        arguments = []
+        selects = []
+        selects_arguments = []
+        for field, value in query_info['params'].items():
+            result = self.process_queried_field(field, value)
+            if result is None:
+                continue
+            sql, value, select = result
+            wheres.append(sql)
+            arguments.append(value)
+            if select is not None:
+                selects.append(select)
+                selects_arguments.append(value)
+
+        return typing.cast(ParsedQueryInfo, dict(
+            query_info,
+            wheres=wheres,
+            wheres_arguments=arguments,
+            selects=selects,
+            selects_arguments=selects_arguments,
+        ))
+
+
+class BasicJsonIndex:
+    operators: typing.List[str] = ['=', '!=', '?', '?|']
+
+    def __init__(self, name: str):
+        self.name = name
+
+    @property
+    def idx_name(self) -> str:
+        return 'idx_objects_{}'.format(self.name)
+
+    @property
+    def index_sql(self) -> typing.List[str]:
+        return [
+            f'''CREATE INDEX CONCURRENTLY IF NOT EXISTS {sqlq(self.idx_name)}
+                ON objects ((json->>'{sqlq(self.name)}'));''',
```

Comment:
I think this should be `CREATE INDEX ... ON {storage._objects_table_name} (...)` (because of supporting custom postgres schemas).

The reason the tests are passing right now even when using `DB_SCHEMA=custom` is because of the function `_clear_dbs()`.
https://github.com/plone/guillotina/blob/a64bc8da76ca71e7c967bd9cd57b51f4bddb1117/guillotina/tests/fixtures.py#L284-L293

This function calls `iter_databases()` and during the process of iterating databases creates a new db 'postgres'. That's because `factory.get_names()` is returning `['postgres', 'guillotina']` (why 'postgres'?) and because 'postgres' is not in `loaded`, `factory.get_database(db_name)` creates the tables `(public.)objects` and `blobs`.

https://github.com/plone/guillotina/blob/4c621a6b782569c9cf23d90325e827491951bdef/guillotina/utils/content.py#L231-L240

You can make tests fail changing `_clear_dbs()` with the following code:
```python
async def _clear_dbs(root):
    # make sure to completely clear db before carrying on...
    for _, db in root:
        if not IDatabase.providedBy(db):
            continue
        storage = db.storage
        ...
```
But i'm not sure if there is a root problem in `iter_databases()` or somewhere else...